mergify_engine/engine/__init__.py

```diff
@@ -232,36 +290,12 @@ def run(client, pull, subscription, sources):
         )
         return
 
-    # CheckRun are attached to head sha, so when user add commits or force push
-    # we can't directly get the previous Mergify Summary. So we copy it here, then
-    # anything that looks at it in next engine runs will find it.
-
-    synchronize_events = dict(
-        (
-            (s["data"]["after"], s["data"])
-            for s in ctxt.sources
-            if s["event_type"] == "pull_request"
-            and s["data"]["action"] == "synchronize"
-        )
-    )
-    if synchronize_events:
-        ctxt.log.debug("engine synchronize summary")
-
-        # NOTE(sileht): We sometimes got many synchronize in a row, that not always the
-        # last one that have the Summary, so we also looks in older one if necessary.
-        after_sha = ctxt.pull["head"]["sha"]
-        while synchronize_events:
-            sync_event = synchronize_events.pop(after_sha, None)
-            if sync_event:
-                if copy_summary_from_previous_head_sha(ctxt, sync_event["before"]):
-                    break
-                else:
-                    after_sha = sync_event["before"]
-            else:
-                ctxt.log.warning(
-                    "Got synchronize event but didn't find Summary on previous head sha",
-                )
-                break
+    ensure_summary_on_head_sha(ctxt)
+    if any(
+        s["event_type"] == "pull_request" and s["data"]["action"] == "closed"
+        for s in ctxt.sources
+    ):
+        actions_runner.delete_last_summary_head_sha(ctxt)
```

Comment:
I think that's fine for now, but I wonder if we wouldn't need a higher abstraction to have such things run properly. Like hooks based on events that you could register. It feels hackish otherwise.

Reply:
I added that as NOTE, can be a good enhancement if we have other think like that.